cloudslang-compiler/src/main/java/io/cloudslang/lang/compiler/modeller/ExecutableBuilder.java

```diff
@@ -478,11 +490,15 @@ private StepModellingResult compileStep(
         if (MapUtils.isEmpty(doRawData)) {
             errors.add(new RuntimeException("Step: \'" + stepName + "\' has no reference information"));
         } else {
-            String refString = doRawData.keySet().iterator().next();
-            refId = resolveReferenceID(refString, imports, namespace);
+            try {
+                String refString = doRawData.keySet().iterator().next();
+                refId = resolveReferenceID(refString, imports, namespace);
+            } catch (RuntimeException rex) {
+                errors.add(rex);
```

Comment:
if all the validators would return a list of exceptions we would not need to do this, but we need to decide on a convention